EnableIPythonDisplay.swift

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Hooks IPython to the KernelCommunicator, so that it can send display
+ * messages to Juptyer.
+ */
+
+import Python
+
+// Workaround SR-7757.
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import func Darwin.C.dlopen
+#else
+import func Glibc.dlopen
+#endif
+dlopen("libpython2.7.so", RTLD_NOW | RTLD_GLOBAL)
+
+var _socket: PythonObject = Python.None
```

Comment:
Why use underscored names?

Reply:
I want to mark them as "system-defined" variables that you shouldn't redeclare -- the REPL lets you redeclare anything, and redeclaring these would mess stuff up. I added a comment.

There might be some way to improve the situation in the future. For example, add source-imports and add a rule that forbits redeclaring things declared by imported modules.

Reply:
You can make them static members of an enum, which is a standard way to introduce a namespace and make them not redeclarable in Swift. That way you can drop the underscore too.

```swift
enum _IPythonDisplayGuts {
  static var socket: PythonObject
  static var shell: PythonObject
}
```

Reply:
ooo, that looks good. done